llvm/lib/CodeGen/InlineSpiller.cpp

```diff
@@ -167,6 +167,10 @@ class InlineSpiller : public Spiller {
   // All registers to spill to StackSlot, including the main register.
   SmallVector<Register, 8> RegsToSpill;
 
+  // All registers that were replaced by the spiller. This includes registers
+  // that were rematerialized; rematted regs are removed from RegsToSpill.
+  SmallVector<Register, 8> RegsReplaced;
```

Comment:
If this is a superset of RegsToSpill, can you just use RegsToSpill and a supplemental set for the rematerialize or split cases?

Reply:
I'm not sure what to return out of the spiller if I do this. It's currently an ArrayRef, but if the result is computed, what should get returned?

Reply:
Return the separate fields?